### DISCUSSION TITLE
캐릭터 기본 애니메이션 설정 방법이 어떻게 될까요?

### DISCUSSION THREAD

**Discussion by ryusiin** [naverz/zepeto-studio-kor#496] [5. Q&A]
목적 : NPC가 NavMesh로 이동할 때 걷기 모션을 적용하고 싶습니다.

알고 싶은 것 : 캐릭터 기본 애니메이션(걷기, 달리기 등)을 보여주는 법을 알고 싶습니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 제페토 애니메이션은 Packages > ZEPETO.Character.Controller > Runtime > _Resources > Animations > base에 기본 애니메이션이 있습니다.
바로 밑에 Gesture 폴더에는 춤, 오케이, 슬립 애니메이션이 있습니다.

![스크린샷 2022-03-15 오후 4 03 40](https://user-images.githubusercontent.com/100118650/158325743-799e3124-ebf8-48f7-bf94-3055d43bd120.png)
 
원하는 타이밍에 원하는 모션을 세팅해서 스크립트를 추가하면 되는데
아래는 제가 버튼을 눌렀을 때 춤을 추는 코드를 예시로 만들었습니다.
참고하셔서 기능을 직접 구현해 보세요 😄

### <스크립트>

import { Canvas, AnimationClip, WaitForSeconds} from 'UnityEngine';
import { ZepetoPlayers, ZepetoCharacter } from 'ZEPETO.Character.Controller';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script';
import { Button } from "UnityEngine.UI";

export default class ShootButton extends ZepetoScriptBehaviour {

    public danceButton : Button;
    private zepetoCharacter: ZepetoCharacter;
    public animationClip : AnimationClip;
    
    Start() {
        //create character
        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this.zepetoCharacter = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character;
        });
        // when button click
        this.danceButton.onClick.AddListener(()=>{
            //start gesture
            this.zepetoCharacter.SetGesture(this.animationClip);
            this.StartCoroutine(this.DoRoutine());
        });
    }
    
    //after 3 seconds later, stop gesture
    *DoRoutine() {
        yield new WaitForSeconds(3);
        this.zepetoCharacter.CancelGesture();
    }
    
}

### <버튼에 스크립트를 적용한 모습>
![스크린샷 2022-03-15 오후 4 14 47](https://user-images.githubusercontent.com/100118650/158327292-1466859e-3836-4c02-9ebc-d8685100e304.png)


### <결과 화면>

https://user-images.githubusercontent.com/100118650/158326674-cb2ff712-93ae-4ec3-970f-9c0a00ae9675.mov

**Reply by hipsc159** [reply]
이 애니메이션을 다른 유저들도 보게하려면 어떻게 해야 하나요?

**Reply by Zepeto-TechEvangelist** [reply]
서버와 클라이언트간의 메세지 송수신을 통해서 구현할 수 있습니다.
플레이어 위치를 갱신하는 아래 가이드를 참고 부탁드리며, 이 부분은 잘 동작하는지 우선 확인 부탁드립니다.
https://studio.zepeto.me/guides/multiplay-room-message
